### ISSUE TITLE
Implicit dimension choice for F.softmax warning

### ISSUE THREAD

**Issue by GlacierMelt** [Shaoli-Huang/SnapMix#5]
UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
